include/mqtt/message.hpp

```diff
@@ -343,13 +342,13 @@ class connect_message {
             connect_flags::set_will_qos(connect_flags_, w.value().qos());
 
             utf8string_check(w.value().topic());
-            will_topic_name_ = as::buffer(w.value().topic());
+            will_topic_name_ = as::buffer(w.value().topic().data(), w.value().topic().size());
```

Comment:
Maybe we should create an overloaded version of as::buffer, that supports mqtt::buffer, directly?

Reply:
I agree with you.
How about this? https://wandbox.org/permlink/pSJtAHVYWniirG7R

Reply:
That looks good to me.

Reply:
Fixed.